include/pstl_offload/internal/usm_memory_replacement_common.h

```diff
@@ -205,125 +242,20 @@ __allocate_shared_for_device(__sycl_device_shared_ptr __device_ptr, std::size_t
     return __ptr;
 }
 
-inline void
-__free_usm_pointer(__block_header* __header)
-{
-    assert(__header != nullptr);
-    __header->_M_uniq_const = 0;
-    sycl::context __context = __header->_M_device.__get_context();
-    __header->~__block_header();
-    sycl::free(__header->_M_original_pointer, __context);
-}
-
-inline void*
-__realloc_real_pointer(void* __user_ptr, std::size_t __new_size)
```

Comment:
Do we move `__realloc_real_pointer` and to the `__aligned_realloc_real_pointer` to cpp file, because they need access to the unordered_map (in case of page+ alignment)?

Reply:
Yes.